#Running to do list
+ Add MuTect2 into Loeb Snakemake workflow
+ Check MuTect2 output for 100% Tumor Colo Sample, and create ROC curve.
    + Work on code to generate the FM/FP plots.
    + Create code to generate ROC curves overall
+ Run cutadapt prior to barcode collapse with BMFtools, and compare to run where cutadapt comes after collapse.
+ Run translocation pipeline on the ctDNA sample
+ Create analysis snakemake workflow
+ Demultiplex Swift HiSeq Run
+ Run pipeline for IDT Inline HiSeq Run (**DONE**)
+ Carefully read CAPP-Seq paper, identify useful methods, assess our current approach.
+ Create new bed file for the new V3 ctDNA/Solid Tumor bed file, adding in probes for Colo829 mutations.

##ctDNA_V3 Capture Bed file
/home/brett/projects/BMFTools_Devel/Panel/V3_Panel

    awk '{print $1"\t"$2"\t"$3}' HOTSPOT_SNV_ROI_gene_cds_v1.bed > hotspot_temp.bed
    awk '{print $1"\t"$2"\t"$3}' designed-probe-coords.bed > temp_designed.bed
    cat temp_desinged.bed hotspot_temp.bed | bedtools sort -faidx ../names.txt -i - > ctDNA_V3_hotspot_SNVs_ROI_with_Colo829_dopes.bed

Total size of capture target is

    cat ctDNA_V3_hotspot_SNVs_ROI_with_Colo829_dopes.bed | awk -F'\t' 'BEGIN{SUM=0}{ SUM+=$3-$2 }END{print SUM}'

**= 57,151kb**

I've placed the combined bed file on the CHPC at:

    /uufs/chpc.utah.edu/common/home/arup-storage1/u0521163/Panel_Development/ctDNA_V3_Panel/ctDNA_V3_hotspot_SNVs_ROI_with_Colo829_dopes.bed

and it lives on my local machine at:

    /home/brett/projects/BMFTools_Devel/Panel/V3_Panel/ctDNA_V3_hotspot_SNVs_ROI_with_Colo829_dopes.bed

All analyses from the new HiSeq runs should be on this file.

##IDT Inline Adapters HiSeq run 12316R
    /uufs/chpc.utah.edu/common/home/arup-storage1/u0521163/ctDNA_Data/12316R_Inline_IDT_HiSeq

I've run all of the samples except NA12878 with the standard Snakemake analysis files, creating a seperate YAML for each pair of analyses.  The Two per yaml is pretty arbitrary, but it's how my workflow is set up for tumor/normal analyses.

    snakemake -np --configfile <paired names>.yaml -j 999 --cluster-config cluster.yaml --cluster "sbatch -A {cluster.account} -p {cluster.partition} -n {cluster.n} -t {cluster.time} -c {cluster.c}" all

is a template command line for how each pair were run on the CHPC.  Going forward, I now need to do variant calling and family, coverage, etc. meta-analysis on these data.

I should also aling the NA12878 to the custom reference I created, and need to place on the CHPC to do so.


##NGS Steering Committee Meeting
In a somewhat contentious meeting, it was highlighted that Pipeline2.0 should be the priority for our department.  However, I don't think this project is being taken on by a single bioinformatician at any step, which is a **huge** problem.  We need to get this assigned to a lead.  Someone who will take responsibility for it.  Terrifying.

##Demultiplex HiSeq Swift Run
After a couple flase starts, these were properly demultiplexed (edit: *Nope*).  Because the swift chem uses a 6 base index barcode, they were not properly being pulled out using the --use-bases-mask argument y133,i6,n*,y133 that I modified from the previous TruSeq runs.  Instead I had to look at the adapter sequences from illumina and manually add the 7th base into the SampleSheet.csv.  **HOWEVER** For some reason only two reads were created for each run.  Dammit.  I'll need to take another stab at this.

#Running to do list
+ Add MuTect2 into Loeb Snakemake workflow
+ Check MuTect2 output for 100% Tumor Colo Sample, and create ROC curve.
    + Work on code to generate the FM/FP plots.
    + Create code to generate ROC curves overall
+ Run cutadapt prior to barcode collapse with BMFtools, and compare to run where cutadapt comes after collapse.
+ Run translocation pipeline on the ctDNA sample
+ Create analysis snakemake workflow
+ Demultiplex Swift HiSeq Run
+ Run analysis pipeline for IDT Inline HiSeq Run
+ Carefully read CAPP-Seq paper, identify useful methods, assess our current approach.
+ Create new bed file for the new V3 ctDNA/Solid Tumor bed file, adding in probes for Colo829 mutations.
