#Running to do list
+ Add MuTect2 into Loeb Snakemake workflow
+ Check MuTect2 output for 100% Tumor Colo Sample, and create ROC curve.
    + Work on code to generate the FM/FP plots.
    + Create code to generate ROC curves overall
+ Run cutadapt prior to barcode collapse with BMFtools, and compare to run where cutadapt comes after collapse.
+ Run translocation pipeline on the ctDNA sample
+ Create analysis snakemake workflow
+ Demultiplex Swift HiSeq Run (**DONE**)
+ Run analysis pipeline for IDT Inline HiSeq Run
+ Run collapse and alignment pipeline on the Swift data
+ Carefully read CAPP-Seq paper, identify useful methods, assess our current approach.
+ Run new Inline barcode samples through MuTect2, in particular the new normals, etc.
+ re-run Strelka on these samples as well, with the new bed file.

##Demultiplex Swift HiSeq run
I solved the problem from demultiplexing the swift run by changing the use bases mask argument in the shell script.

    /mnt/raw1/HiSeq_BMF/bcl2fastq2_BMF.v2.16.secondary.sh

from

    y137,i7,n*,y137
to

    y137,i7,y*,y137

I'm not sure why it was ever 'n' to begin with, shouldn't it always be a data read?  Strange.  I'm a bit worried about what that meant for previous runs.

Looking back at the last run it looks like I was still using bcl2fastq2 1.83 or whatever that was, which produced a make file and a bit more stats.  However, now running with 2.13 everything seems to work with that new --use-bases-mask as above.  

Reads are now in

    /mnt/fastq1/HiSeq_BMF/160324_D00784_0034_BHJJ2CBCXX

and will be transfered to the CHPC shortly.

##IDT Inline barcodes HiSeq run

    /uufs/chpc.utah.edu/common/home/arup-storage1/u0521163/ctDNA_Data/12316R_Inline_IDT_HiSeq

All of the bams are completed, aligned to b37.  However, I did attempt to mess with the NA12878.yaml and NA12878.sm to get it aligned to the NA12878 reference, however some components of the pipeline won't be able to run (specfically the GATK portions) because of the differences in the coordinates between the NA12878 ref and the noraml b37 ref.  This run has finished as well and the results are in

   /uufs/chpc.utah.edu/common/home/arup-storage1/u0521163/ctDNA_Data/12316R_Inline_IDT_HiSeq/rescued_reads/NA12878

The alignment to the NA12878 reference is used for the per-base error corrections.

##Analysis Snakemake pipeline
These are analyses that I need to put into the snakemake Analysis
+ Coverage plots, both for cumulative coverage, and per-probe coverage
+ Family size histograms
+ Depth of Coverage, GATK's or BMFTools
+ NA12878 error rates by family size (probably not in the snakemake script)
+ Maybe MuTect and Strelka?

##Meta-analysis of IDT Inline HiSeq Run
I'm pulling down the IDT inline hiseq run bams to my local machine to make some plots, do variant calling, etc.  I'll also start a new Google sheet to share the meta-analysis of these data.

The connection speeds are keeping me slow here.  Waiting on data to move to and from CHPC, both from ngs-prod3 and to my local machine.

##Running cutadapt before bardcode collapse
    /uufs/chpc.utah.edu/common/home/arup-storage1/u0521163/ctDNA_Data/HiSeq_Loeb_Colo_Run2_wCutAdapt/cut_adapt_first

I'm going to run this using my older Snakemake pipeline.

#Running to do list
+ Add MuTect2 into Loeb Snakemake workflow
+ Check MuTect2 output for 100% Tumor Colo Sample, and create ROC curve.
    + Work on code to generate the FM/FP plots.
    + Create code to generate ROC curves overall
+ Run cutadapt prior to barcode collapse with BMFtools, and compare to run where cutadapt comes after collapse.
+ Run translocation pipeline on the ctDNA sample
+ Create analysis snakemake workflow
+ Run analysis pipeline for IDT Inline HiSeq Run
+ Run collapse and alignment pipeline on the Swift data
+ Carefully read CAPP-Seq paper, identify useful methods, assess our current approach.
+ Run new Inline barcode samples through MuTect2, in particular the new normals, etc.
+ re-run Strelka on these samples as well, with the new bed file.
