#Running to do list
+ Add MuTect2 into Loeb Snakemake workflow
+ Check MuTect2 output for 100% Tumor Colo Sample, and create ROC curve.
    + Work on code to generate the FM/FP plots.
    + Create code to generate ROC curves overall
+ Run cutadapt prior to barcode collapse with BMFtools, and compare to run where cutadapt comes after collapse.
+ Run translocation pipeline on the ctDNA sample
+ Create analysis snakemake workflow
+ Run analysis pipeline for IDT Inline HiSeq Run
+ Run collapse and alignment pipeline on the Swift data
+ Carefully read CAPP-Seq paper, identify useful methods, assess our current approach.
+ Run new Inline barcode samples through MuTect2, in particular the new normals, etc.
+ re-run Strelka on these samples as well, with the new bed file.

##BMFTools
Updated to the 1.0 branch on both the CHPC and my local machine.  Hope this doesn't break anything or have wildly different results.

##Swift Collapsing and Aligning
    /uufs/chpc.utah.edu/common/home/arup-storage1/u0521163/ctDNA_Data/12318R_9N_Swift_HiSeq

The Swift libraries require only minor changes to the config file for snakemake, using the secondary index workflow.  Mainly adding

    -s 3
    -t 7

to the sdmp command line should be enough.  I also removed this line

    #configfile: "tumor_normal_config.yaml"

So the config yaml would be passed from the command line.

All of the runs have been started on the CHPC, but there are abundant jobs already running.  Hopefull they'll sneak in overnight.

I'll need to set up the special run for NA12878 tomorrow morning.

##Lunch Meeting with Barry Moore
We discussed the nature of continued collaborations between the UCGD/UGP group and ARUP's Biocomputing department.  He recommended we use the WGS pipeline development to open and maintain cooperation between the groups.  We could learn abundantly from their WGS stuff (fastQforward) and would also like to add VAAST/pVAAST/Phevor into our workflows.  The ladder will require substantial political wrangling.   

He also recommended that we be sure to use the wgs to perhaps open a discussion on comparing our variant calling pipelines across the board, and to use our operational/infrastructure resources and expertise to help them expand their abilities into the cloud.  Specifically this will probably require some interaction between the infrastructre/ops group (Mark's group) and Carson and Shawn.  Getting FastQForward into our production cloud will be considerable work, but will benefit both parties.  

I mentioned that we are looking to recruit people and without poaching from them directly.  He agreed to send anyone that is looking for industry positions my way.

#Running to do list
(somehow this keeps getting longer and not shorter)
+ Add MuTect2 into Loeb Snakemake workflow
+ Check MuTect2 output for 100% Tumor Colo Sample, and create ROC curve.
    + Work on code to generate the FM/FP plots.
    + Create code to generate ROC curves overall
+ Run cutadapt prior to barcode collapse with BMFtools, and compare to run where cutadapt comes after collapse.
+ Run translocation pipeline on the ctDNA sample
+ Create analysis snakemake workflow
+ Run analysis pipeline for IDT Inline HiSeq Run
+ Run collapse and alignment pipeline on the Swift data (**done for all but NA12878**)
+ Carefully read CAPP-Seq paper, identify useful methods, assess our current approach.
+ Run new Inline barcode samples through MuTect2, in particular the new normals, etc.
+ re-run Strelka on these samples as well, with the new bed file.
